### PR TITLE
prevent db connections during asset precompile

### PR DESCRIPTION
**Pull request by karlhungus**
#### Problem

We are creating db connections during asset compilation

#### Solution

Do connections on load

Full disclosure: i have no idea what i'm doing :(

@kirs @rafaelfranca /cc @celsodantas @fw42

### CODE REVIEW

File: lib/bugsnag/railtie.rb [hunk 1]
```diff
@@ -45,8 +45,10 @@ class Railtie < Rails::Railtie
         ActionController::API.send(:include, Bugsnag::Rails::ControllerMethods)
       end
       if defined?(ActiveRecord::Base)
-        require "bugsnag/rails/active_record_rescue"
-        ActiveRecord::Base.send(:include, Bugsnag::Rails::ActiveRecordRescue)
+        ActiveSupport.on_load(:active_record) do
```
Comment (karlhungus) [inline lib/bugsnag/railtie.rb:48]: @celsodantas pointed out this is probably unacceptable because it creates a dependency on active support :(.

**Comment by rafaelfranca** [comment]
This is inside a railtie and then already depends on Active support. This
is the correct way to do this.

**Comment by karlhungus** [comment]
going to close this and point it at bugsnag/master -- i'll reping you all

**Comment by kirs** [comment]
We can't use bugsnag master right now because it has code that's not compatible with Shopify.

https://github.com/Shopify/shopify/pull/93214

**Comment by kirs** [comment]
Ok, it should be fine with Frenzy 👌

**Comment by karlhungus** [comment]
bugsnag/bugsnag-ruby version: https://github.com/bugsnag/bugsnag-ruby/pull/337

**Comment by karlhungus** [comment]
@kirs assuming it's ok :cool: if you merge this to [bugsnag/bugsnag-ruby](https://github.com/bugsnag/bugsnag-ruby/pull/337) (i don't have write access) 

Edit: forgot link.

**Pull request closed without merging**